### PR TITLE
Taxon range improvements

### PR DESCRIPTION
**Pull request by loarie**
made changes described here https://github.com/inaturalist/inaturalist/issues/3212

### CODE REVIEW

**Comment by loarie** [comment]
@kueda I added taxon_range/show to this pull request and I think its ready for re-review. If you could also give a shot at adding a link from the taxon maps to taxon_range/show that would be awesome
Screen Shot 2021-09-08 at 7 03 48 PM

File: app/views/taxon_ranges/_form_fields.html.haml
Comment (kueda) [inline app/views/taxon_ranges/_form_fields.html.haml:24]: If you use `BootstrapFormBuilder` instead of `DefaultFormBuilder` you can re-write this as `= f.radio_button :iucn_relationship, 2, label: t(:taxon_not_in_the_iucn_redlist)`, plus you'll get the default styles for the other form fields. The source choose will cause problems, though. You might have to hard code the width of that input for it to work properly.

**Review by kueda — CHANGES_REQUESTED** [review]

File: config/locales/en.yml
Comment (kueda) [inline config/locales/en.yml:2105]: Again, I would avoid embedding HTML in the translatable text because any changes in markup will require re-translation, and translators tend to make mistakes with the markup. I think it also helps to name view-specific keys like this with some consistent prefix so they stay together in the source file, so like

```yaml
taxon_range_guidelines_if_not_uploading: "If you are not uploading an IUCN RedList range map, this should be because:"
taxon_range_guidelines_iucn_unsuitable: IUCN RedList range map for this taxon is unsutable
```

etc. Also, FWIW, have a translation for bold text followed by a colon: `bold_label_colon_value_html`.

File: app/views/taxon_ranges/_form_fields.html.haml
Comment (kueda) [inline app/views/taxon_ranges/_form_fields.html.haml:15]: Since `iucn_relationship` can be null, I think we need a fourth option here for that, like "other" or "unknown" or something. In addition to "unknown" can't a range be for a taxon that *is* on the Red List but the range didn't come from the IUCN?

File: app/views/taxon_ranges/show.html.haml
Comment (kueda) [inline app/views/taxon_ranges/show.html.haml:3]: Needs to be globalized.

File: app/views/taxon_ranges/show.html.haml
Comment (kueda) [inline app/views/taxon_ranges/show.html.haml:21]: I don't think this needs to block merge, but this is a bad habit: images that the app uses should be checked into the repo and served via the asset pipeline.

File: config/locales/en.yml
Comment (kueda) [inline config/locales/en.yml:236]: Translation keys that contain HTML should end in `_html`. That way you don't have to use `raw` when you use them in the templates.

File: config/locales/en.yml
Comment (kueda) [inline config/locales/en.yml:2188]: "RedList" should be "Red List"

File: config/locales/en.yml
Comment (kueda) [inline config/locales/en.yml:2200]: "RedList" should be "Red List"

File: app/models/taxon_range.rb
Comment (kueda) [inline app/models/taxon_range.rb:10]: This... isn't actually what this validation is doing, right?

File: app/models/taxon_range.rb
Comment (kueda) [inline app/models/taxon_range.rb:8]: This results in an error like "Taxon taxon range already exists." I would leave the message blank, or change it to "already has a range" so the message reads "Taxon already has a range" instead. Actually, this text should be globalized, but I'm not actually sure how to do that properly off the top of my head.

File: config/locales/en.yml
Comment (kueda) [inline config/locales/en.yml:2681]: Definitely good to translate these errors, but this translation will not be used when you do `errors.add(:iucn_relationship, :must_be_an_accepted_value)`. For that to work it needs to exist at `activerecord.errors.models.taxon_range.attributes.iucn_relationship.must_be_an_accepted_value`

File: config/locales/en.yml
Comment (kueda) [inline config/locales/en.yml:3770]: Proposed re-write:

> Please use the description to explain how the map was derived and link to any relevant flags, taxonomic splits, or other discussions of why the IUCN Red List range required alteration.

File: config/locales/en.yml
Comment (kueda) [inline config/locales/en.yml:4696]: "RedList" should be "Red List"

File: config/locales/en.yml
Comment (kueda) [inline config/locales/en.yml:4762]: "RedList" should be "Red List"

File: app/assets/stylesheets/taxon_ranges/taxon.css
Comment (kueda) [inline app/assets/stylesheets/taxon_ranges/taxon.css:1]: Since this has nothing to do with the taxon the range is associated with, how about renaming this file to `form.css`. Also needs a newline at the end.

**Review by kueda — CHANGES_REQUESTED** [review]

**Comment by loarie** [comment]
@kueda I think this is ready to merge if you're happy with it

File: app/views/taxon_ranges/_form_fields.html.haml
Comment (loarie) [reply app/views/taxon_ranges/_form_fields.html.haml:15]: My vision was that we should work towards every taxon range having one of these 3 values:
IUCN_REDLIST_MAP
IUCN_REDLIST_MAP_HAS_ISSUES
NOT_ON_IUCN_REDLIST
So while we display unknown for the many existing records where this is unknown, we want to encourage curators to choose one of those 3 if they are going to update that attribute.
For the range is on the Red List but the range didn't come from the IUCN case it would be IUCN_REDLIST_MAP_HAS_ISSUES because there's still some reason why someone felt the IUCN range was unsuitable to display and therefore opted for an alternative.

Comment (kueda) [reply app/views/taxon_ranges/_form_fields.html.haml:15]: Ah, that wasn't clear to me. Could we change that text to read, "This is not an IUCN Red List range map because that map has problems" or something? Maybe the last could be "This is not an IUCN Red List range map because this taxon is not on the IUCN Red List"?

Comment (loarie) [reply app/views/taxon_ranges/_form_fields.html.haml:15]: I changed it to that but I'm going with "unsuitable" rather than "has problems" to make it more subjective to iNat's purposes

**Review by kueda — APPROVED** [review]

**Pull request merged**
